### PR TITLE
feat: add unitest for models

### PR DESCRIPTION
**Pull request by koxudaxi**

### CODE REVIEW

File: lilypad/server/models/spans.py [hunk 1]
```diff
@@ -71,7 +71,8 @@ def _convert_span_table_to_public(
         elif span.scope == Scope.LLM:
             data = span.data
             display_name = f"{data['attributes']['gen_ai.system']} with '{data['attributes']['gen_ai.request.model']}'"
-        else:
+        else: # pragma: no cover
+            # This should never happen. Because the scope is an Enum, it should always be one of the two values. And, SQLModel should enforce this.
```
Comment (willbakst) [inline lilypad/server/models/spans.py:75]: if this is the case, why don't we just put the above `elif` block inside an `else` block instead?

File: tests/server/test_models.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    SpanPublic,
+    SpanTable,
+    VersionCreate,
+    VersionPublic,
+    VersionTable,
+)
+
+
+@pytest.fixture
+def engine():
```
Comment (willbakst) [inline tests/server/test_models.py:32]: nit: rename to `fixture_engine` so it's clear it's a fixture

File: tests/server/test_models.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    """Create an in-memory database engine for testing."""
+    engine = create_engine(
+        "sqlite://", connect_args={"check_same_thread": False}, poolclass=StaticPool
+    )
+    SQLModel.metadata.create_all(engine)
+    return engine
+
+
+@pytest.fixture
+def session(engine):
```
Comment (willbakst) [inline tests/server/test_models.py:42]: nit: same here, `fixture_session`

File: tests/server/test_models.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    """Create an in-memory database engine for testing."""
+    engine = create_engine(
+        "sqlite://", connect_args={"check_same_thread": False}, poolclass=StaticPool
+    )
+    SQLModel.metadata.create_all(engine)
+    return engine
+
+
+@pytest.fixture
+def session(fixture_engine):
```
[the remark this answers is not in the document]
Comment (willbakst) [reply tests/server/test_models.py:42]: Circling back on this, I think this should actually just be `session` and the above should just be `engine`

For fixtures that use mocks, we should prepend mock e.g. `mock_session`, but this particular engine and session are not actually mocks they are real, so using mock would be confusing.

File: tests/server/test_models.py [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+    SpanPublic,
+    SpanTable,
+    VersionCreate,
+    VersionPublic,
+    VersionTable,
+)
+
+
+@pytest.fixture
+def fixture_engine():
```
Comment (willbakst) [inline tests/server/test_models.py:32]: let's actually change this back to just `engine`

Comment (willbakst) [reply tests/server/test_models.py:32]: The convention should be:
1. Use `mock_` when the fixture actually mocks something (e.g. fake data, MagicMock, etc.)
2. Use clear naming for anything that is not a mock (e.g. `sqlite_engine` in this case)

Comment (willbakst) [inline tests/server/test_models.py:32]: this is missing a type hint return annotation

File: tests/server/test_models.py (hunk 4 above)
Comment (willbakst) [inline tests/server/test_models.py:42]: this is missing a type hint on the engine arg

File: tests/server/test_models.py [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+@pytest.fixture
+def session(fixture_engine):
+    """Create a new database session for testing."""
+    with Session(fixture_engine) as session:
+        yield session
+        session.rollback()
+
+
+def test_base_sql_model():
```
Comment (willbakst) [inline tests/server/test_models.py:49]: these are all missing return annotations of `None` please update all tests to return `None`

**Review by willbakst — APPROVED** [review]
